scico/operator/_operator.py

```diff
@@ -47,7 +49,11 @@ def _wrap_mul_div_scalar(func: Callable) -> Callable:
 
     @wraps(func)
     def wrapper(a, b):
-        if np.isscalar(b) or isinstance(b, jax.core.Tracer):
+        if (
+            snp.isscalar(b)
+            or isinstance(b, jax.core.Tracer)
+            or (isinstance(b, jaxlib.xla_extension.ArrayImpl) and b.size == 1)
```

Comment:
I prefer `b.ndim==0` to `b.size==1`.

Reply:
This is an interesting one. In principle an array with unit size and dimension can also function as a scalar in multiplication and division. If `ndim` is 0 the pseudo-scalar `dtype` does not affect the result `dtype`, but if `ndim` is 1 it does, which greatly complicates the implementation, so I agree that it's better to base the test on `ndim` to exclude this case.